### PR TITLE
misc: prepare for mpich-4.0a2

### PR DESCRIPTION
**Pull request by hzhou**
## Pull Request Description
Update CHANGES, RELEASE_NOTES, etc.

## Author Checklist
* [x] **Provide Description** 
      Particularly focus on _why_, not _what_. Reference background, issues, test failures, xfail entries, etc.
* [x] **Commits Follow Good Practice**
      Commits are self-contained and do not do two things at once. 
      Commit message is of the form: `module: short description` 
      Commit message explains what's in the commit.
* [x] **Passes All Tests**
      Whitespace checker. Warnings test. Additional tests via comments.
* [x] **Contribution Agreement**
      For non-Argonne authors, check [contribution agreement](http://www.mpich.org/documentation/contributor-docs/). 
      If necessary, request an explicit comment from your companies PR approval manager.

### CODE REVIEW

**Comment by hzhou** [comment]
test:mpich/ch4/most
test:mpich/ch3/most

File: configure.ac [hunk 1]
```diff
@@ -5353,11 +5353,18 @@ AC_CONFIG_FILES([Makefile \
 AC_OUTPUT
 
 if test ${device_name} = "ch4" ; then
+    t_netmod=$ch4_netmods
+    if test "$ch4_netmods" = "ofi" -a "$with_libfabric" = "embedded"; then
+        t_netmod="ofi (using embedded libfabric)"
+    elif test "$ch4_netmods" = "ucx" -a "$with_libfabric" = "embedded"; then
+        t_netmod="ucx (embedded)"
```
Comment (raffenet) [inline configure.ac:5360]: should be checking `$with_ucx` in the conditional

File: CHANGES [hunk 2]
```diff
@@ -2,23 +2,16 @@
...
-# Implemented basic MPI-4 session API using MPIX interface.
+# All MPI-4 APIs have been implemented. Major MPI-4 features include MPI
+  sessions, partitioned point-to-point communications, events in the MPI tool
+  information interface, large-count functions, persistent collectives,
+  MPI_Comm_idup_with_info, MPI_Isendrecv and MPI_Isendrecv_replace,
+  MPI_Info_get_string, MPI_Comm_split_type with new split_type --
+  MPI_COMM_TYPE_HW_GUIDED and MPI_COMM_TYPE_HW_UNGUIDED.
 
-# Implemented basic MPI-4 partitioned communication API using MPIX interface.
+# Add QMPI support.
```
Comment (raffenet) [inline CHANGES:12]: Might want to note that this is an experimental feature.

**Review by raffenet — COMMENTED** [review]
Looks good, just a couple minor issues.

File: configure.ac (hunk 1 above)
Comment (hzhou) [reply configure.ac]: Thanks. Fixed.

File: CHANGES (hunk 2 above)
Comment (hzhou) [reply CHANGES]: Changed to `Add QMPI (experimental) support.`

**Review by raffenet — APPROVED** [review]

**Pull request merged**